relecov_tools/read_bioinfo_metadata.py

```
@@ -410,31 +423,44 @@ def handling_tables(self, file_list, conf_tab_name):
         file_ext = os.path.splitext(conf_tab_name)[1]
-        # Parsing key position
         sample_idx_colpos = self.get_sample_idx_colpos(self.current_config_key)
         extdict = {".csv": ",", ".tsv": "\t", ".tab": "\t"}
-        if file_ext in extdict.keys():
-            data = relecov_tools.utils.read_csv_file_return_dict(
-                file_name=file_list[0],
-                sep=extdict.get(file_ext),
-                key_position=sample_idx_colpos,
-            )
-            return data
-        elif conf_tab_name.endswith(".gz"):
-            self.log_report.update_log_report(
-                method_name,
-                "warning",
-                f".gz files are not supported yet for data extraction: {conf_tab_name}",
-            )
-            data = {}
-        else:
-            self.log_report.update_log_report(
-                method_name,
-                "error",
-                f"Unrecognized defined file name extension '{file_ext}' in '{conf_tab_name}'.",
-            )
-            sys.exit(self.log_report.print_log_report(method_name, ["error"]))
-        return data
+        mapping_fields = self.software_config[self.current_config_key].get("content")
+
+        if not mapping_fields:
+            return {}
+
+        if conf_tab_name.endswith(".gz"):
+            inner_ext = os.path.splitext(conf_tab_name[:-3])[1]
+            if inner_ext in extdict:
+                self.log_report.update_log_report(
+                    method_name,
+                    "warning",
+                    f"Expected tabular file '{conf_tab_name}' is compressed and cannot be processed.",
+                )
+            return {}
+
+        if file_ext in extdict:
+            try:
+                return relecov_tools.utils.read_csv_file_return_dict(
+                    file_name=file_list[0],
+                    sep=extdict[file_ext],
+                    key_position=sample_idx_colpos,
+                )
+            except Exception as e:
```

If you check the code for  `read_csv_file_return_dict()` you might find that it should only raise `FileNotFoundError`. The generic `Exception` might hide an error that should be addressed correctly in the future.